File: snippets/java/app/src/main/java/finetuning/CreateFinetunedModel.java
/* (C)2024 */
package finetuning;

import com.cohere.api.Cohere;
import com.cohere.api.resources.finetuning.finetuning.types.*;

public class CreateFinetunedModel {
    public static void main(String[] args) {
        Cohere cohere = Cohere.builder().clientName("snippet").build();

        CreateFinetunedModelResponse response =
                cohere.finetuning()
                        .createFinetunedModel(
                                FinetunedModel.builder()
                                        .name("test-finetuned-model")
                                        .settings(
                                                Settings.builder()
                                                        .baseModel(
                                                                BaseModel.builder()
                                                                        .baseType(
                                                                                BaseType
                                                                                        .BASE_TYPE_CHAT)
                                                                        .build())
                                                        .datasetId("my-dataset-id")
                                                        .build())
                                        .build());

        System.out.println(response);
    }
}

File: snippets/node/finetuning/create-finetuned-model.js
import { Cohere, CohereClient } from 'cohere-ai';

const cohere = new CohereClient({ token: process.env.COHERE_API_KEY! });

(async () => {
  const finetunedModel = await cohere.finetuning.createFinetunedModel({
    name: 'test-finetuned-model',
    settings: {
      base_model: {
        base_type: Cohere.Finetuning.BaseType.BaseTypeChat,
      },
      dataset_id: 'test-dataset-id',
    },
  });

  console.log(finetunedModel);
})();
